numpy/core/src/umath/reduction.c

```diff
@@ -293,98 +68,86 @@ count_axes(int ndim, const npy_bool *axis_flags)
  *            better quality error messages. For example, "numpy.max"
  *            would be a good name for NumPy's max function.
  *
- * Returns a view which contains the remaining elements on which to do
- * the reduction.
+ * Returns 0 on success and -1 if an error occurred.
  */
-NPY_NO_EXPORT PyArrayObject *
+NPY_NO_EXPORT int
```

Comment:
Slightly bigger: how about making this a function that just returns a view of the first element? (i.e., returns `op_view`) It feels that the copy at the end doesn't really belong inside here, but rather should be done in the caller.

Reply:
Opted for renaming the function for now, not sure what is nicer.

Reply:
Returning the size makes sense.